Sources/CvPlot.cpp

```diff
@@ -710,31 +710,43 @@ void CvPlot::doTurn()
 }
 
 
+// Returns odds to be inverted and checked against for a given bonus and improvement
+int CvPlot::calculateResourceDiscoveryOdds(int iBonus, CvImprovementInfo pInfo) const
+{
+	int iOdds = pInfo.getImprovementBonusDiscoverRand(iBonus);
+	
+	iOdds *= GC.getGameSpeedInfo(GC.getGame().getGameSpeedType()).getConstructPercent();
+	iOdds /= 100;
+
+	// Bonus density normalization (use land tiles, account for bonus type?)
+	iOdds *= 7 * (GC.getMap().getNumBonuses((BonusTypes) iBonus) + 2);
+	iOdds /= 2 * (GC.getMap().getWorldSize() + 9);
+	
+	return iOdds;
+}
+
+
 void CvPlot::doImprovement()
```

Comment:
Forgot I had this commit from earlier; seems to work fine, but maybe have a more experienced cpp person glance at it to make sure before v41.1?